onnxruntime/python/tools/transformers/fusion_qordered_layernorm.py

```diff
@@ -0,0 +1,95 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation.  All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from logging import getLogger
+from typing import Dict
+
+from fusion_base import Fusion
+from fusion_utils import FusionUtils
+from onnx import helper
+from onnx_model import OnnxModel
+
+logger = getLogger(__name__)
+
+
+class FusionQOrderedLayerNormalization(Fusion):
+    def __init__(self, model: OnnxModel):
+        super().__init__(model, "QOrderedLayerNormalization", "LayerNormalization")
+
+    def fuse(self, node, input_name_to_nodes: Dict, output_name_to_node: Dict):
```

Comment:
Consider using Mapping for the annotation, here and elsewhere

```suggestion
    def fuse(self, node, input_name_to_nodes: Mapping[str, Any], output_name_to_node: Mapping[str, Any]):
```

If we know how the nodes are typed, I would also replace Any with that

-----

Use more generic types for input parameters and specific types for return values. For example, a function may take a `Sequence` and return a `List`.

Diagram for reference: https://gist.github.com/justinchuby/4021cebe9e093f636759a88de325c85f

Reply:
Thanks for this education. I will read more about this.